File: pkg/models/orientation.go
package models

type OrientationEnum string

const (
	OrientationLandscape OrientationEnum = "LANDSCAPE"
	OrientationPortrait  OrientationEnum = "PORTRAIT"
	OrientationSquare    OrientationEnum = "SQUARE"
)

func (e OrientationEnum) IsValid() bool {
	switch e {
	case OrientationLandscape, OrientationPortrait, OrientationSquare:
		return true
	}
	return false
}

File: ui/v2.5/src/utils/orientation.ts
import { OrientationEnum } from "src/core/generated-graphql";

const stringOrientationMap = new Map<string, OrientationEnum>([
  ["Landscape", OrientationEnum.Landscape],
  ["Portrait", OrientationEnum.Portrait],
  ["Square", OrientationEnum.Square],
]);

export const stringToOrientation = (
  value?: string | null,
  caseInsensitive?: boolean
) => {
  if (!value) {
    return undefined;
  }

  const ret = stringOrientationMap.get(value);
  if (ret || !caseInsensitive) {
    return ret;
  }

  const asUpper = value.toUpperCase();
  const foundEntry = Array.from(stringOrientationMap.entries()).find((e) => {
    return e[0].toUpperCase() === asUpper;
  });

  if (foundEntry) {
    return foundEntry[1];
  }
};

export const orientationStrings = Array.from(stringOrientationMap.keys());
